packages/@lwc/template-compiler/src/__tests__/fixtures/directive-for-each/children/expected.js

```diff
@@ -1,5 +1,39 @@
 import { registerTemplate } from "lwc";
 function tmpl($api, $cmp, $slotset, $ctx) {
+  const { items: $cv0_0 } = $cmp;
+  function foreach4_0(item) {
+    return api_element(
+      "div",
+      {
+        key: api_key(7, item.id),
+      },
+      []
+    );
+  }
+  function if2_1() {
+    function foreach3_0(item) {
+      return [
+        api_element(
+          "p",
+          {
+            key: api_key(3, item.id),
+          },
+          [api_text("X1")]
+        ),
+        api_element(
+          "p",
+          {
+            key: api_key(4, item.id),
+          },
+          [api_text("X2")]
+        ),
+      ];
+    }
+    return api_iterator($cv0_0, foreach3_0);
+  }
+  function foreach1_0(item) {
+    return api_text("X");
+  }
```

Comment:
Nested if block and for each block makes it complex/impossible to understand the actual code flow.

One way to make it more intelligible would be to use a similar approach than Svelte with comments preceding blocks. Example of [generated code](https://svelte.dev/repl/e9f0854138b44e8c82c7bc5278ddda08?version=3.38.3).